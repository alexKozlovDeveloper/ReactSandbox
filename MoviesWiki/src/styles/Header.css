.header {
    background-color: #7f8eba;
}

.headertitle {
    padding: 40px;
    font-family: cursive;
    color: aliceblue;
}