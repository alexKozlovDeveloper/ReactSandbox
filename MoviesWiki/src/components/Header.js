import React, { Component } from "react";

import Search from "./Search";

import styles from "../styles/Header.css"

class Header extends Component {
    render() {
        return (
            <div className={styles.header}>
                <div className={styles.container}>
                    <div className={styles.headertitle}>Find Your Movies</div>
                    <Search />                    
                </div>
            </div>
        );
    }
}

export default Header;