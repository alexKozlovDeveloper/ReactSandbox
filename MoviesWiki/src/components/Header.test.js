import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import Header from "./Header";

configure({adapter: new Adapter()});

test('Header rendering test', () => {

    var headerConfig = {
        title: "Find Your Movies",
        searchConfig: {
            placeHolder: "Quentin Tarantino",
            buttonText: "Search",
            searchFilter: {
                title: "Search by",
                selectedIndex: "0",
                options: ["Title", "Genre"]
            }
        }
    }

    const component = shallow(<Header config={headerConfig}/>);
    expect(shallowToJson(component)).toMatchSnapshot();
 });

