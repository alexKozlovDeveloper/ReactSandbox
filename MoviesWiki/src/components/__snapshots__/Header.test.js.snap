// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header rendering test 1`] = `
<div
  className="header"
>
  <div
    className="container"
  >
    <div
      className="headertitle"
    >
      Find Your Movies
    </div>
    <withRouter(Connect(Search))
      config={
        Object {
          "buttonText": "Search",
          "placeHolder": "Quentin Tarantino",
          "searchFilter": Object {
            "options": Array [
              "Title",
              "Genre",
            ],
            "selectedIndex": "0",
            "title": "Search by",
          },
        }
      }
    />
  </div>
</div>
`;
